emulator/blockchain.go

```diff
@@ -1335,6 +1348,35 @@ func (b *Blockchain) commitBlock() (*flowgo.Block, error) {
 
 	// reset pending block using current block and ledger state
 	b.pendingBlock = newPendingBlock(block, ledger, b.clock)
+	b.entropyProvider.LatestBlock = block.ID()
+
+	// lastly we execute the system chunk transaction
+	txn, err := b.systemChunkTransaction()
+	if err != nil {
+		return nil, err
+	}
+	ctx := fvm.NewContextFromParent(
+		b.vmCtx,
+		fvm.WithAuthorizationChecksEnabled(false),
+		fvm.WithSequenceNumberCheckAndIncrementEnabled(false),
+		fvm.WithRandomSourceHistoryCallAllowed(true),
+	)
+
+	executionSnapshot, output, err := b.vm.Run(
+		ctx,
+		fvm.Transaction(txn, uint32(len(transactions))),
+		b.pendingBlock.ledgerState,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	b.pendingBlock.events = append(b.pendingBlock.events, output.Events...)
+
+	err = b.pendingBlock.ledgerState.Merge(executionSnapshot)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
extract this into a function, something like updateRandomSourceHistory

Reply:
Good point :+1: 
Updated in https://github.com/onflow/flow-emulator/pull/511/commits/e1686844a60e3165344b08bb9be89fa695a5bf26